components/Alert.tsx

```diff
@@ -1,101 +1,81 @@
-            </div>
-        ), 
-        warning: (
-            <div
-                className="bg-orange-200 px-6 py-3 my-4 rounded-md flex items-center fixed w-1/2"
-                >
-                <svg
-                    viewBox="0 0 24 24"
-                    className="text-yellow-600 w-5 h-5 sm:w-5 sm:h-5 mr-3"
-                    >
-                    <path
-                        fill="currentColor"
-                        d="M23.119,20,13.772,2.15h0a2,2,0,0,0-3.543,0L.881,20a2,2,0,0,0,1.772,2.928H21.347A2,2,0,0,0,23.119,20ZM11,8.423a1,1,0,0,1,2,0v6a1,1,0,1,1-2,0Zm1.05,11.51h-.028a1.528,1.528,0,0,1-1.522-1.47,1.476,1.476,0,0,1,1.448-1.53h.028A1.527,1.527,0,0,1,13.5,18.4,1.475,1.475,0,0,1,12.05,19.933Z"
-                        ></path>
-                </svg>
-                <span className="text-yellow-800">{ alert ? alert.message : undefined }</span>
-            </div>
-        ),
-        info: (
-            <div
-                className="bg-blue-200 px-6 py-3 my-4 rounded-md flex items-center fixed w-1/2 top-0"
-                >
-                <svg
-                    viewBox="0 0 24 24"
-                    className="text-blue-600 w-5 h-5 sm:w-5 sm:h-5 mr-3"
-                    >
-                    <path
-                        fill="currentColor"
-                        d="M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm.25,5a1.5,1.5,0,1,1-1.5,1.5A1.5,1.5,0,0,1,12.25,5ZM14.5,18.5h-4a1,1,0,0,1,0-2h.75a.25.25,0,0,0,.25-.25v-4.5a.25.25,0,0,0-.25-.25H10.5a1,1,0,0,1,0-2h1a2,2,0,0,1,2,2v4.75a.25.25,0,0,0,.25.25h.75a1,1,0,1,1,0,2Z"
-                        ></path>
-                </svg>
-                <span className="text-blue-800">{ alert ? alert.message : undefined }</span>
-            </div>
-        ),
-        success: (
-            <div
-                className="bg-green-200 px-6 py-3 my-4 rounded-md flex items-center fixed w-1/2"
-                >
-                <svg
-                    viewBox="0 0 24 24"
-                    className="text-green-600 w-5 h-5 sm:w-5 sm:h-5 mr-3"
-                    >
-                    <path
-                        fill="currentColor"
-                        d="M12,0A12,12,0,1,0,24,12,12.014,12.014,0,0,0,12,0Zm6.927,8.2-6.845,9.289a1.011,1.011,0,0,1-1.43.188L5.764,13.769a1,1,0,1,1,1.25-1.562l4.076,3.261,6.227-8.451A1,1,0,1,1,18.927,8.2Z"
-                        ></path>
-                </svg>
-                <span className="text-green-800">{ alert ? alert.message : undefined }</span>
-            </div>
-        )
-    }
+    const [displayAlert, setDisplayAlert] = useState<AlertType>()
 
+    // only called on mount; thus ignoring lint warning of including alert in dependency
     useEffect(() => {
         if (alert) {
-            setDisplayAlert(alerts[alert.type])
+            setDisplayAlert(alert)
+            setAlert(undefined);        // resets the global context immediately
+        }
+    }, [setAlert])
```

Comment:
on mount, if an alert is present, we would want to display it immediately no matter the value of displayNow (as opposed to the second useEffect), bc it means we pushed to another page with an alert

but after mounting, we differentiate between messages we want to display on the same page (e.g. login field not filled in correctly) or after we push to another page (e.g. order created successfully after coming back to home screen) 

setAlert is just to prevent the linter from complaining lol, it doesn't affect anything since it doesn't change after mount (defined in alertContext)